lib/api/api.go

```diff
@@ -484,7 +498,7 @@ func (s *service) CommitConfiguration(from, to config.Configuration) bool {
 	// No action required when this changes, so mask the fact that it changed at all.
 	from.GUI.Debugging = to.GUI.Debugging
 
-	if to.GUI == from.GUI {
+	if cmp.Equal(to.GUI, from.GUI) {
```

Comment:
Yes, this discussion is duplicated in https://github.com/emlun/syncthing/pull/6#discussion_r1730328307 . Shall I take `reflect.DeepEqual` as the resolution in both these cases, then?

Reply:
As I understand the comment from @imsodin, `reflect.DeepEqual` is the accepted solution, so that meta PR can be merged and the same applied here as well.

Reply:
Reacted on that other comment thread, before seeing the string equality option here. I think both reflect.DeepEqual and the string method should be ok, definitely preferable to cmp.

Reply:
Fixed in caad2ff7b8ef3fd8a49055650ea0b3b1b444f666 and 32cb11cb90c23f9c8c0ee3f72c0f6342096df326. Some uses of `cmp.Equal` remain in tests where `reflect.DeepEqual` doesn't work, presumably because of the difference in how they compare nil slices/structs. Is that okay?